packages/eslint-plugin/src/rules/no-unnecessary-qualifier.ts

```diff
@@ -93,9 +93,8 @@ export default createRule({
         accessedSymbol.flags,
         context.sourceCode.getText(name),
       );
-      return (
-        fromScope === undefined || symbolsAreEqual(accessedSymbol, fromScope)
-      );
+
+      return !!fromScope && symbolsAreEqual(accessedSymbol, fromScope);
```

Comment:
```
// If the symbol in scope is different, the qualifier is necessary.
```
According to the comment (in line 89), if a symbol is not in scope, it should not be considered unnecessary. If you have a counterexample, please let me know.